qa/L0_trace/test.sh

```diff
@@ -691,6 +691,132 @@ if [ `grep -o 'parent_id' bls_trace.log | wc -l` != "2" ]; then
     RET=1
 fi
 
+# Attempt to trace non-existent model
+SERVER_ARGS="--trace-file=trace_off_to_min.log --trace-level=OFF --trace-rate=1 --model-control-mode=explicit --model-repository=$MODELSDIR"
+SERVER_LOG="./inference_server_off.log"
+run_server
+if [ "$SERVER_PID" == "0" ]; then
+    echo -e "\n***\n*** Failed to start $SERVER\n***"
+    cat $SERVER_LOG
+    exit 1
+fi
+
+# Explicitly load model
+rm -f ./curl.out
+set +e
+code=`curl -s -w %{http_code} -o ./curl.out -X POST localhost:8000/v2/repository/models/simple/load`
+set -e
+if [ "$code" != "200" ]; then
+    cat ./curl.out
+    echo -e "\n***\n*** Test Failed\n***"
+    RET=1
+fi
```

Comment:
I think a lot of lines in this file can be condensed with some helper functions. Something along these lines, but you can get creative with it:
```bash
# Helpers
function assert_curl_success {
  message="${1}"
  if [ "$code" != "200" ]; then
    cat ./curl.out
    echo -e "\n***\n*** ${message}\n***"
    RET=1
  fi
}

function assert_curl_failure {
  message="${1}"
  if [ "$code" != "400" ]; then
    cat ./curl.out
    echo -e "\n***\n*** ${message}\n***"
    RET=1
  fi
}

function get_trace_setting {
  model_name="${1}"
  rm -f ./curl.out
  set +e
  code=`curl -s -w %{http_code} -o ./curl.out localhost:8000/v2/models/${model_name}/trace/setting`
  set -e
}

function update_trace_setting {
  model_name="${1}"
  settings="${2}"
  rm -f ./curl.out
  set +e
  code=`curl -s -w %{http_code} -o ./curl.out -X POST localhost:8000/v2/models/${model_name}/trace/setting -d ${settings}`
  set -e
}

...

# Assert trace endpoints return errors for non-existent models
GOOD_MODEL="simple"
BAD_MODEL="does-not-exist"
TRACE_SETTINGS='{"log_frequency": "1"}'

get_trace_setting ${BAD_MODEL}
assert_curl_failure "Expected non-existent model to fail"

update_trace_setting "${BAD_MODEL}" "${TRACE_SETTINGS}" 
assert_curl_failure "Expected non-existent model to fail"

# Assert trace endpoints succeed for existing model
load_model ${GOOD_MODEL}
get_trace_setting ${GOOD_MODEL}
assert_curl_success "Expected existing model to succeed"

update_trace_setting "$GOOD_MODEL}" "${TRACE_SETTINGS}" 
assert_curl_success "Expected existing model to succeed"

unload_model ${GOOD_MODEL}
...
...
```

Can be applicable to some existing parts of the test you didn't touch as well